Documentation/README.md

```diff
@@ -1,6 +1,26 @@
 Documents Index
 ===============
 
-- [Developer Guide](developer-guide.md)
+## Overview and general information
+
 - [Intro to .NET Core CLI](intro-to-cli.md)
```

Comment:
I see a rename below that doesn't seem to match this. I prefer this location for this document.

This section can probably be deleted: https://github.com/dotnet/cli/blob/master/Documentation/intro-to-cli.md#historical-context---dnx